### ISSUE TITLE
Relations to fields other than only PK (orm: db.go)

### ISSUE THREAD

**Issue by bartek5186** [beego/beego#3921]
TLDR;
There are no possibility to connect and fill relations based on FK with fields other than PK.

Beego 1.12.0
GoVer 1.10.4
AMD64, linux

3. What did you do?

Two tables with relations
```
type User struct {
Id int `orm:"column(id);pk"` // normal PK
Uuid  string `orm:"column(uuid);type(uuid);null;unique"`
}
```

Second Table with rel(fk):
```
type PassengerRequests struct {
TakenBy  *User  `orm:"column(taken_by);type(uuid);null;rel(fk)"` // related to UUID from User
}
```

4. What did you expect to see?

When creating relation, code always expose to PK field, but sometimes there are need relation
to another fields - that are not PK, but for example FK with other datatype like UUID (not only depends on int type, in this scenario we use, a UUID field, not PK Integer).

In our example the databases relations are build between taken_by (uuid) and uuid (uuid) from another model. Normal in DB this scenario is easy and possible, but in beego not. Because in beego these are always related to PK (not FK, or other type than int)

5. What did you see instead?

I notice that, beego always build a FK based on "fk" string writed in Struct.
In db.go code:

```
case fieldType&IsRelField > 0:
  fi = fi.relModelInfo.fields.pk
  fieldType = fi.fieldType
  goto setValue
}
```
**fi = fi.relModelInfo.fields.pk**
This code looks that is connected to this issue. There are always "pk". But we need relation to another field that - for example FK without remove PK  from Struct (id).

In model - there You can also config another indexes, maybe there is a good place to define this:
```
// multiple fields index
func (t *User) TableIndex() [][]string {
	return [][]string{
		[]string{"Id", "Uuid"},
	}
}
```

Nice way will be a possibility to use "fk" string in Structs, to connect between models on other indexes than primary key.
